openedx/features/colaraz_features/api/serializers.py

```diff
@@ -2,69 +2,106 @@
+    create_oauth2_client_for_ecommerce_site,
+    create_site_on_ecommerce,
+    get_existing_site_domains,
+    get_or_create_organization,
+    get_or_create_sites,
+    update_or_create_site_configurations,
+    update_or_create_site_themes,
 )
 
 
 class SiteOrgSerializer(serializers.Serializer):
     site_domain = DomainField(max_length=90)
     site_name = serializers.CharField(max_length=40, required=False, default='')
-    site_theme = serializers.CharField(max_length=255, default=settings.DEFAULT_SITE_THEME)
+    site_theme = serializers.CharField(max_length=255, required=False, default=settings.DEFAULT_SITE_THEME)
     org_name = serializers.CharField(max_length=255)
-    org_short_name = serializers.SlugField(max_length=255)
     platform_name = serializers.CharField(max_length=255)
+    company_type = serializers.CharField(max_length=255, required=False, default='')
     university_name = serializers.CharField(max_length=255, required=False, default='')
     organizations = serializers.ListField(
         child=serializers.SlugField(max_length=255)
     )
+    site_partner = serializers.CharField(max_length=255, required=False, default='edx')
+    payment_processors = serializers.CharField(max_length=255, required=False, default='')
+    client_side_payment_processor = serializers.CharField(max_length=255, required=False, default='')
+    ecommerce_from_email = serializers.EmailField(max_length=255, required=False, default=None)
+    payment_support_email = serializers.EmailField(max_length=255, required=False, default=None)
 
-    @staticmethod
-    def validate_site_domain(value):
+    def validate(self, fields):
         """
-        Validate that site domain does not already exists.
+        Validate all the required fields.
         """
-        if do_sites_exists(domain=value):
-            raise serializers.ValidationError('An LMS or studio site with domain "{}" already exists.'.format(value))
-        return value
+        self.sites_validation(fields.get('site_name'), fields.get('site_domain'))
+        self.org_validation(fields.get('org_name'))
+        return fields
 
     @staticmethod
-    def validate_org_name(value):
+    def sites_validation(name, domain):
         """
-        Validate that organization name does not already exists.
+        Validate that site domains does not already exists.
         """
-        if Organization.objects.filter(name=value).exists():
-            raise serializers.ValidationError('An organization with name "{}" already exists.'.format(value))
-        return value
+        site_domains = get_existing_site_domains(name=name, domain=domain)
+        if site_domains:
+            raise serializers.ValidationError(
+                'Sites with domains "{}" already exist.'.format([str(i) for i in site_domains])
+            )
 
     @staticmethod
-    def validate_org_short_name(value):
+    def org_validation(value):
```

Comment:
The old name seems better. of `validate_organization`. but up to you to decide

Reply:
basically those were default names of validators that run automatically while I want to handle them by myself.